activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -170,8 +170,11 @@ def migration_context # :nodoc:
       class Version
         include Comparable
 
-        def initialize(version_string)
+        attr_reader :full_version_string
+
+        def initialize(version_string, full_version_string = nil)
           @version = version_string.split(".").map(&:to_i)
+          @full_version_string = full_version_string
         end
```

Comment:
Wait, are we storing this Version object in the schema cache? We'll have to make sure that a schema cache from 5.2 can still be read on 6.0 and vice versa. Also across adapters.

Reply:
Also, since you added the version caching, how does other adapters take part in that caching? — and make sure that they don't break the schema cache read with broken YAML. cc @yahonda

Reply:
Each adapter is responsible for what gets stored in the schema cache.

https://github.com/rails/rails/blob/713f62430a42294dcb156cc8f38b41509a60d7a6/activerecord/lib/active_record/connection_adapters/schema_cache.rb#L96-L98

For example,  the PosgreSQL adapter will use the version string.

https://github.com/rails/rails/blob/713f62430a42294dcb156cc8f38b41509a60d7a6/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L424-L427

And the MySQL adapter will store the `Version` object.

https://github.com/rails/rails/blob/713f62430a42294dcb156cc8f38b41509a60d7a6/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L58-L60

There is also a test that verifies we can read a 5.1 schema cache dump in Rails 6.

https://github.com/rails/rails/blob/713f62430a42294dcb156cc8f38b41509a60d7a6/activerecord/test/cases/connection_adapters/schema_cache_test.rb#L60-L72

Reply:
> Also, since you added the version caching, how does other adapters take part in that caching? — and make sure that they don't break the schema cache read with broken YAML. cc @yahonda

Thanks for the heads-up. 
https://github.com/rsim/oracle-enhanced/pull/1859 has been taken care of to support #35795